src/containers/TransactionPage/TransactionPage.test.js

```diff
@@ -1171,6 +1171,90 @@ describe('TransactionPage', () => {
       });
     });
 
+    it('Inquiry with restricted view rights should not show OrderPanel - hour unit', async () => {
+      const config = {
+        ...getHostedConfiguration(),
+        accessControl: { marketplace: { private: true } },
+      };
+
+      const routeConfiguration = getRouteConfiguration(config.layout);
+      const processName = 'default-booking';
+      const bookingInquiry = createInquiry(processName, 'hour', bookingTransitions.INQUIRE);
+      const transactionRole = TX_TRANSITION_ACTOR_CUSTOMER;
+      const nextTransitions = [
+        {
+          id: new UUID('next-transition'),
+          type: 'processTransition',
+          attributes: {
+            name: 'transition/request-payment-after-inquiry',
+            actor: [transactionRole],
+            actions: [],
+            params: {},
+          },
+        },
+      ];
+
+      const stateData = getStateData(
+        {
+          transaction: bookingInquiry,
+          transactionRole,
+          nextTransitions,
+          intl: fakeIntl,
+          transitionInProgress: false,
+          transitionError: null,
+          onTransition: noop,
+          sendReviewInProgress: false,
+          sendReviewError: null,
+          onOpenReviewModal: noop,
+        },
+        getProcess(processName)
+      );
+
+      const currentUser = createCurrentUser(customerId);
+      currentUser.effectivePermissionSet.attributes.read = 'permissions/deny';
+
+      const props = {
+        ...panelBaseProps,
+        currentUser,
+        transaction: bookingInquiry,
+        transactionRole,
+        nextTransitions,
+        params: {
+          id: bookingInquiry.id?.uuid,
+        },
+        lineItems: [],
+      };
+
+      const { getAllByText } = render(<TransactionPageComponent {...props} />, {
+        config,
+        routeConfiguration,
+      });
+
+      await waitFor(() => {
+        const state = stateData.processState;
+        const txTitle = `TransactionPage.${processName}.${transactionRole}.${state}.title`;
+        expect(screen.getByText(txTitle)).toBeInTheDocument();
+
+        expect(screen.getByText('TransactionPage.listingTitleMobile')).toBeInTheDocument();
+
+        // Activity feed (here we just check the heading)
+        expect(screen.getByText('TransactionPanel.activityHeading')).toBeInTheDocument();
+
+        // Show listing's title only (side card)
+        expect(getAllByText('listing-item title')).toHaveLength(1);
+
+        // Don't show listing pricing
+        expect(screen.queryByText('$55.00')).toBeNull();
+        expect(screen.queryByText('OrderPanel.perUnit')).toBeNull();
+
+        // Don't show BookingTimeForm inputs
+        expect(screen.queryByText('BookingTimeForm.bookingStartTitle')).toBeNull();
```

Comment:
Side note: I haven't used `.toBeNull()` expectation. So, another approach is `.not.toBeInTheDocument()`

Reply:
I originally tried with `.not.toBeInTheDocument()`, but I kept getting errors that the test engine wasn't able to find the item, i.e. the page was behaving correctly but the test expectation wasn't catching it correctly. On the other hand, the combination of queryByText and toBeNull resolved correctly.

Reply:
Looks like the issue was with screen.getByText vs screen.queryByText and not the .not.toBeInTheDocument vs .toBeNull. So I could update these to `expect(screen.queryByText(...)).not.toBeInTheDocument()`!

Reply:
@SariSaar I think this is fine. 
It's just more dramatic to be able to ask the big question: "toBe or not.toBe?"